specutils/spectra/spectral_coordinate.py

```diff
@@ -458,7 +459,12 @@ def _calculate_radial_velocity(observer, target):
 
         d_vel = target_icrs.velocity - observer_icrs.velocity
 
-        return d_vel.d_xyz * pos_hat.xyz
+        vel_mag = np.dot(d_vel.d_xyz, pos_hat.xyz)
+
+        if as_scalar:
+            return vel_mag
+        else:
+            return vel_mag * pos_hat.xyz
```

Comment:
The previous implementation was incorrect - one wants to return a vector that has the total velocity magnitude times the unit vector between observer and target.